### PR TITLE
Added Shuffle String Solution

### PR DESCRIPTION
**Pull request by manishh12**
## Related Issue

Fix : #438 

## Description
Added solution for kShuffle string under leetcode solution section.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/1500-1599/1528-Shuffle-String.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+By using both simple iteration and map-based approaches, we can efficiently solve the Shuffle String problem. The choice between the two approaches depends on the specific requirements and constraints of the problem.
+
+:::
+
+## References
+
+- **LeetCode Problem:** [Shuffle String](https://leetcode.com/problems/shuffle-string/)
+- **Solution Link:** [Shuffle String Solution on LeetCode](https://leetcode.com/problems/shuffle-string/solution/)
+- **Authors LeetCode Profile:** [Manish Kumar Gupta](https://leetcode.com/_manishh12/)
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1500-1599/1528-Shuffle-String.md:331]: remove line or code `---`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
remove last or end line `---`

**Comment by manishh12** [comment]
@Ajay-Dhangar Done kindly review it.

**Review by Ajay-Dhangar — APPROVED** [review]
done

**Pull request merged**